server/app/views/questiontypes/ApplicantQuestionRendererParams.java

```diff
@@ -30,14 +30,31 @@ public enum ErrorDisplayMode {
      * Review page. Typically used when the applicant has clicked Review but their answers didn't
      * pass validation.
      */
-    DISPLAY_ERRORS_WITH_MODAL_REVIEW;
+    DISPLAY_ERRORS_WITH_MODAL_REVIEW,
+    /**
+     * Validation errors are displayed, *and* a modal is displayed on top of the form asking the
+     * applicant to either (1) correct the errors or (2) discard their answers and continue to the
+     * previous block. Typically used when the applicant has clicked Previous but their answers
+     * didn't pass validation.
+     */
+    DISPLAY_ERRORS_WITH_MODAL_PREVIOUS;
 
     /**
      * Returns true if the given mode indicates that validation errors should be rendered when
      * displaying the form and false otherwise.
      */
     public static boolean shouldShowErrors(ErrorDisplayMode mode) {
-      return mode == DISPLAY_ERRORS || mode == DISPLAY_ERRORS_WITH_MODAL_REVIEW;
+      return mode == DISPLAY_ERRORS
+          || mode == DISPLAY_ERRORS_WITH_MODAL_REVIEW
+          || mode == DISPLAY_ERRORS_WITH_MODAL_PREVIOUS;
+    }
+
+    /**
+     * Returns ture if the given mode indicates that validation errors *and* a modal should be
```

Comment:
s/ture/true/

Also can we format this comment as a javadoc? (I'm surprised it didn't yell at you about is already.)

Reply:
Wait is this not formatted as Javadoc haha?

Reply:
I had been thinking it also needs the like `@returns` line, but if nothing else in this file is following that I don't feel all that strongly about starting now.

Reply:
Oh gotcha! Javadoc style guide recommends that you start with a summary fragment that's a full sentence, and then the extra things like `@returns` are only added if they're necessary -- https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment.